collector/lib/SysdigService.cpp

```diff
@@ -223,6 +226,24 @@ void SysdigService::Start() {
   running_ = true;
 }
 
+void LogUnreasonableEventTime(int64_t time_micros, sinsp_evt* evt) {
+  int64_t time_diff;
+  int64_t evt_ts = evt->get_ts() / 1000UL;
+  int64_t max_past_time = 3600000000;  // One hour in microseconds. This is generous.
+  int64_t max_future_time = 5000000;   // 5 seconds in microseconds. A time in a little in the future is suspicious.
+
+  time_diff = time_micros - evt_ts;
+  if (time_diff > max_past_time) {
+    CLOG_THROTTLED(WARNING, std::chrono::seconds(10)) << "Event of type " << evt->get_type() << " is unreasonably old. It's timestamp is " << google::protobuf::util::TimeUtil::MicrosecondsToTimestamp(evt_ts);
+    COLLECTOR_ASSERT(false);
```

Comment:
Perhaps the logging statement and assert could be combined. See above. Perhaps something other than false could be passed in, like `(time_diff > max_past_time)`.

Reply:
I have removed the assert.